IQSS/dataverse
# remove NONE as a license option for SWORD, show available licenses #8551

**What this PR does / why we need it**:

The example XML in the SWORD docs doesn't work due to licenses changes in 5.10 and PR #7920.

**Which issue(s) this PR closes**:

- Closes #8551

**Special notes for your reviewer**:

I added some minimal testing of interactions between license and rights. We could add more.

I'm not 100% sure of all the assertions in the SWORD page of the API Guide. We could add more tests and/or edit the guide as needed.

Generally speaking, I'm thinking we should drop NONE as an option, even if it's an incompatible change. If we want to keep NONE, what should it map to? Custom Dataset Terms? But they can't be blank, right?

**Suggestions on how to test this**:

- Try passing NONE. You should now see a list of possible licenses.
- Try passing a valid license.
- Test all the assertions in the SWORD section of the API Guide about license, rights, etc.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

No.

**Is there a release notes update needed for this change?**:

Maybe?

**Additional documentation**:

None.
